proposals/0003.md

```diff
@@ -0,0 +1,122 @@
+
+In this section we include the templates that need to be included and also other
+recommendations to make the licensing experience as smooth as possible for both
+contributors and core developers.
+
+## License File
+
+Every repository MUST include a `LICENSE` file which corresponds to this
+[file][license-file]. As well as this, there MUST be a `COPYING` file which is a
+symlink to the `LICENSE` file.
+
+## DCO File
+
+Every repository MUST include a `DCO` which corresponds to this
+[file][dco-file].
+
+## License Header
+
+Every file that contains source code MUST include a header which corresponds to
+this [file][license-header] at the **top** of each source file as a comment.
+
+**Note** that there are placeholders `<project name>` which correspond to the
+name of the repository. You MUST replace these with the actual name.
+
+Also included is a [script][license-header-check] for checking the
+license-header for Rust files.
+
+## CONTRIBUTING File
+
+Every repository MUST include a `CONTRIBUTING.md` file. The
+inclusion of the `CONTRIBUTING.md` file will mean that GitHub will prompt new
+contributors to look at the file before making a pull request.
+
+The file must include the following section:
+
+```
+## Certificate of Origin
+
+By contributing to this project, you agree to the [Developer Certificate of
+Origin (DCO)](./path/to/DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution.
+
+In order to show your agreement with the DCO you should include at the end of the commit message,
+the following line: Signed-off-by: John Doe <john.doe@example.com>, using your real name and email.
+
+This can be done easily using the -s flag on the git commit.
+
+If you find yourself pushed a few commits without Signed-off-by, you can still add it afterwards.
+
+### Fixing the DCO
+
+If you did not sign one or more of your commits then it is not all for not. The
+crowd at `src-d` have a wonderful guide on how to remedy this situation
+[here](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)
+
+## License Header
+
+As part of our license, we must include a license header at the top of each
+source file. The template for this header can be found
+[here](./path/to/license-header). If you are creating a new file in the
+repository you will have to add this header to the file.
+```
+
+## DCO Bot
+
+To ensure that all commits by an outside contributor are signed we SHALL add a
+[DOC bot][dco-bot] to monitor pull requests on our repositories.
+
+## Merging Pull Requests
+
+We SHALL NOT merge pull requests using the GitHub UI. This is due to GitHub
+creating commits on our behalf using a bot. These commits will not be signed.
+
+It is RECOMMENDED that the repository is set up with a rebasing strategy to
+ensure the history is only made up of signed commits.
```

Comment:
What does this mean? Are you referring to the “Merge Button” setting of a Github repo or a branch protection rule? Does this even apply if we don’t merge via the Github UI?

Reply:
Ya, so we don't merge via the button. But I was thinking in the case that somebody _does_ press the button then if it's a rebase then there's no extra commit.

Reply:
So you’re referring to the “Allow rebase merging” “Merge Button” setting. Even with that the history will not be made up of signed commits because Github will create new commits for every commit in the PR.

Reply:
I've tested it and the "rebase merge" on GitHub is just broken. Even if you merge a fast-forward branch, it will still rebase it with its dirty unsigned fingers.

Reply:
Ach! Thanks for checking. One other idea is to require 6 maintainers to approve which we rarely get :joy:

Reply:
As per my other [comment](https://github.com/radicle-dev/radicle-decisions/pull/5/files#r531633613), I am not against allowing squash commits via THE GREEN BUTTON. There, the button-operator will have the opportunity to review the commit message (just make sure you don't add a `Signed-off-by` on behalf of someone else lol).

The only reason to not use THE GREEN BUTTON at all is not wanting GitHub as a contributor.

Reply:
Oh wait, scrap that. A squash merge _MUST_ be signed-off. As it's not you, but GitHub, who introduces a _new_ `tree`, you would need to add _your_ `Signed-off-by` in the textbox. At this point, the CLI will win in terms of convenience.